src/pydargs/__init__.py

```diff
@@ -111,15 +131,22 @@ def _create_parser(tp: Type[Dataclass], **kwargs: Any) -> ArgumentParser:
             elif origin in UNION_TYPES:
                 union_parser = partial(_parse_union, union_type=field.type)
                 setattr(union_parser, "__name__", repr(field.type))
-                parser.add_argument(
+                parser_or_group.add_argument(
                     *arguments,
                     type=union_parser,
                     **argument_kwargs,
                 )
             else:
                 raise NotImplementedError(f"Parsing into type {origin} is not implemented.")
+        elif hasattr(field.type, "__dataclass_fields__"):
+            if positional:
+                raise ValueError("Dataclasses may not be positional arguments.")
+            if field.default_factory is not None and field.default_factory != field.type:
+                warn(f"Non-standard default of field {field.name} is ignored by pydargs.", UserWarning)
```

Comment:
This refers to the default provided in e.g. SubConfig(a=1) right? 
```suggestion
                warn(f"Non-standard default for nested dataclass {field.name} is ignored by pydargs.", UserWarning)
```

Reply:
It does, indeed. I don't agree that this suggestion improves the warning; for the user the "nested" may not be clear. In any case the dataclass in question is just a field of the dataclass the user is parsing, so that should be clear.